RELEASE.md

```diff
@@ -190,6 +190,20 @@ The release should be announced on the following social channels:
 - Twitter: Short and sweet in a single tweet, TODO: get Lodestar account.
 - Reddit: TODO: get Lodestar account.
 
+# Release Manager Checklist
+This section is to guide the Release Manager tasked with the next version release to ensure all items have been completed.
+- Confirm consensus on `unstable` release candidate commit
+- Complete Step 1: Create release candidate
+- Complete Step 2: Tag release candidate
+- Deploy `rc.x` candidate to `beta` group of servers
+- If there are `rc.x` hot fixes, push to branch and increment the `rc.x` version.
+- After 3-day testing period, start thread on communication channels for Release Candidate Metrics Review
+- When given the go-ahead, complete Step 4: Merge release candidate
```

Comment:
This step requires others to approve the open PR, probably should mention that somehow / be a little more specific on thie step.